File: Data Structures & Algorithms/largest-rectangle-in-histogram/submission-0.py
class Solution:
    def largestRectangleArea(self, heights: List[int]) -> int:
        max_area = 0
        n = len(heights)

        for i in range(n):
            current_height = heights[i]


            left = i
            while left > 0 and heights[left-1] >= current_height:
                left -= 1
            

            right = i
            while right < n-1 and heights[right+1] >= current_height:
                right += 1
            
            width = right - left + 1

            max_area = max(max_area, current_height*width)
        
        return max_area

File: Data Structures & Algorithms/max-water-container/submission-0.java
class Solution {
    public int maxArea(int[] heights) {
        int left = 0;
        int right = heights.length-1;
        int maxArea = 0;

        while (left<right){
            int minHeight = (heights[left]<heights[right]? heights[left]: heights[right]);
            int area = (right-left)*minHeight;

            maxArea = (maxArea>area?maxArea:area);

            if(heights[left]<heights[right]){
                left++;
            }else{
                right--;
            }
        }
        return maxArea;
    }
}
